test/gtest/ucp/test_ucp_am.cc

```diff
@@ -1020,7 +1020,65 @@ UCS_TEST_P(test_ucp_am_nbx_align, multi)
                       UCP_AM_FLAG_PERSISTENT_DATA);
 }
 
-UCP_INSTANTIATE_TEST_CASE(test_ucp_am_nbx_align)
+class test_ucp_am_nbx_seg_size : public test_ucp_am_nbx {
+public:
+    test_ucp_am_nbx_seg_size() : m_size(0ul)
+    {
+        modify_config("ADDRESS_VERSION", "v2");
+        modify_config("RNDV_THRESH", "inf");
+    }
+
+    void init()
+    {
+        m_size               = ucs_max(UCS_KBYTE,
+                                       ucs::rand() % (64 * UCS_KBYTE));
+        std::string str_size = ucs::to_string(m_size);
+
+        test_ucp_am_nbx::init();
+
+        // Create new sender() with different segment size
+        m_env.push_back(new ucs::scoped_setenv("UCX_IB_SEG_SIZE", str_size.c_str()));
+        m_env.push_back(new ucs::scoped_setenv("UCX_MM_SEG_SIZE", str_size.c_str()));
+        m_env.push_back(new ucs::scoped_setenv("UCX_SCOPY_SEG_SIZE", str_size.c_str()));
```

Comment:
maybe better do it by push_config() + modify_config() + pop_config()

Reply:
but these are uct vars in ucp test

Reply:
i think it should work now, we support UCT config from UCP